PythonTool/TagExpansion.py

```
@@ -139,39 +119,79 @@ def tag_expansion(mesh_file, output_file = None):
-            data_type = tag.get_data_type()
-            scalar_tag = mb_ref.tag_get_handle(tag.get_name(), 1, data_type,
-                                               types.MB_TAG_SPARSE, create_if_missing = True)
-            mb_ref.tag_set_data(scalar_tag, hexes_ref, scalar_data)
-
-        # Check if the user supplied a base output file name.
-        if(output_file):
-            file_location = os.getcwd() + "/" + dir_name + "/" + output_file + str(index) + ".vtk"
-        else:
-            file_location = os.getcwd() + "/" + dir_name + "/" + tag_name + str(index) + ".vtk"
+        data = mb_exp.tag_get_data(vec_tag, hexes)
+        scalar_data = np.copy(data[:,index])
+        data_type = vec_tag.get_data_type()
+        scalar_tag = mb_ref.tag_get_handle(name, 1, data_type, types.MB_TAG_SPARSE,
+                                           create_if_missing = True)
+        mb_ref.tag_set_data(scalar_tag, hexes, scalar_data)
+
+        # Write the mesh file with the new scalar tag.
+        file_location = os.getcwd() + "/" + dir_name + "/" + name + str(index) + ".vtk"
         mb_ref.write_file(file_location)
         index += 1
 
     print(str(index) + " files have been written to disk.")
 
 
+def load_mesh(mesh_file):
+    """
+    Load the mesh file and extract the lists of scalar and vector tags, then
+    expand each vector tag.
+
+    Input:
+    ______
+       mesh_file: str
+           User supplied mesh file location.
+
+    Returns:
+    ________
+       none
+    """
+
+    # Load the mesh file to create a reference mesh.
+    mb_ref = core.Core()
+    mb_ref.load_file(mesh_file)
+
+    # Retrieve the lists of scalar and vector tags on the mesh.
+    hexes_ref, scal_tags_ref, vec_tags_ref = get_tag_lists(mb_ref, types.MBHEX)
+
+    # Make sure the mesh file contains at least one vector tag.
+    if len(vec_tags_ref) < 1:
+        print("WARNING: This mesh file did not contain any vector tags.")
+        exit()
```

here can also raise an error rather than a print statement